dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/MetadataDumperTest.java

```diff
@@ -25,10 +33,123 @@
 @RunWith(JUnit4.class)
 public class MetadataDumperTest {
 
+  private File file;
+  private MetadataDumper dumper = new MetadataDumper().withExitOnError(false);
+  Connector connector = new HiveMetadataConnector();
+
+  @After
+  public void tearDown() throws IOException {
+    if (file == null || !file.exists()) return;
+    if (file.isDirectory()) FileUtils.deleteDirectory(file);
+    else file.delete();
+  }
+
   @Test
   public void testInstantiate() throws Exception {
-    MetadataDumper dumper = new MetadataDumper();
-    // dumper.run("--help");    // Calls System.exit(1)
+    dumper = new MetadataDumper();
     dumper.run("--connector", new BigQueryLogsConnector().getName(), "--dry-run");
   }
+
+  @Test
+  public void testCreatesDefaultOutputZip() throws Exception {
+    // Arrange
+    String name = connector.getDefaultFileName(false);
+    file = new File(name);
+
+    // Act
+    dumper.run("--connector", connector.getName());
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testCreatesDefaultOutputZipInProvidedDirectory() throws Exception {
+    // Arrange
+    String name = connector.getDefaultFileName(false);
+    String path = "dir";
+    file = new File(path, name);
+
+    // Act
+    dumper.run("--connector", connector.getName(), "--output", path);
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testCreatesZipWithGivenName() throws Exception {
+    // Arrange
+    String name = "test.zip";
+    file = new File(name);
+
+    // Act
+    dumper.run("--connector", connector.getName(), "--output", name);
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testOverridesZipWithGivenName() throws Exception {
+    // Arrange
+    String name = "test-dir/test.zip";
+    file = new File(name);
+
+    // Act
+    Files.createParentDirs(file);
+    file.createNewFile();
+    dumper.run("--connector", connector.getName(), "--output", name);
+
+    // Assert
+    Assert.assertTrue(file.exists());
```

Comment:
This assertion would pass even if `dumper.run` is removed. I would suggest to check that the file was overwritten, e.g. by writing a string to a file and checking that the contents is different. If you prefer, possibly remove this test from this PR and add it in a separate PR, in order to merge this PR faster.

Reply:
Nice catch! adding a todo